### PR TITLE
Removed use of deprecated cgi module

### PR DESCRIPTION
**Pull request by andamian**
Fix for #382 and a bug.

### CODE REVIEW

File: pyvo/dal/mimetype.py [hunk 1]
```diff
@@ -75,32 +75,37 @@ def mime_object_maker(url, mimetype, session=None):
...
     session = use_session(session)
-    full_type, params = cgi.parse_header(mimetype)
-    mimetype = [x.strip() for x in full_type.split('/')] if '/' in full_type \
+    m = Message()
+    m['content-type'] = mimetype
+    pp = m.get_params()
+    full_type = pp[0][0]
+    params = pp[1:]
+    mtype = [x.strip() for x in full_type.split('/')] if '/' in full_type \
         else None
```
Comment (bsipocz) [inline pyvo/dal/mimetype.py:84]: We allow longer lines
```suggestion
    mtype = [x.strip() for x in full_type.split('/')] if '/' in full_type else None
```

File: pyvo/dal/mimetype.py [hunk 2]
```diff
@@ -75,32 +75,37 @@ def mime_object_maker(url, mimetype, session=None):
     if not mimetype:
         raise ValueError('mimetype required')
     session = use_session(session)
-    full_type, params = cgi.parse_header(mimetype)
-    mimetype = [x.strip() for x in full_type.split('/')] if '/' in full_type \
+    m = Message()
+    m['content-type'] = mimetype
+    pp = m.get_params()
```
Comment (bsipocz) [inline pyvo/dal/mimetype.py:80]: I prefer meaningful variable names, but as these are not really used later on, I suppose they could stay as is

File: pyvo/dal/mimetype.py [hunk 3]
```diff
@@ -75,32 +75,37 @@ def mime_object_maker(url, mimetype, session=None):
...
         # things start to get messy
-        if mimetype[2].get('content', None) == 'datalink':
-            from .adhoc import DatalinkResults
-            return DatalinkResults.from_result_url(url)
-        else:
-            from .query import DALResults
-            return DALResults.from_result_url(url)
+        for param in params:
+            if (param[0].lower() == 'content') and \
+                    (param[1].lower() == 'datalink'):
```
Comment (bsipocz) [inline pyvo/dal/mimetype.py:107]: these could also fit in one line

**Review by bsipocz — APPROVED** [review]
Looks all good to me, thanks!

**Comment by andamian** [comment]
@bsipocz - this now contains a codestyle fix for the CI as well and I think we can merge it.

**Comment by bsipocz** [comment]
Thanks for fixing that unrelated style issue, I suppose the newer versions of linters became a bit more picky.

**Pull request merged**